File: tests/fixtures/greeter.java
public class Greeter {
    private final String greeting;

    public Greeter(String greeting) {
        this.greeting = greeting;
    }

    public String greet(String name) {
        return greeting + ", " + name + "!";
    }

    public static String freeFunction(String name) {
        return "Hello, " + name;
    }

    public static void main(String[] args) {
        Greeter g = new Greeter("Hello");
        System.out.println(g.greet("world"));
    }
}

File: tests/fixtures/greeter.go
package main

import "fmt"

type Greeter struct {
	greeting string
}

func NewGreeter(greeting string) *Greeter {
	return &Greeter{greeting: greeting}
}

func (g *Greeter) Greet(name string) string {
	return fmt.Sprintf("%s, %s!", g.greeting, name)
}

func freeFunction(name string) string {
	return fmt.Sprintf("Hello, %s", name)
}

func main() {
	g := NewGreeter("Hello")
	fmt.Println(g.Greet("world"))
}
